torch_xla/core/dynamo_bridge.py

```diff
@@ -126,18 +128,36 @@ def _args_on_cuda(input_args: tuple) -> bool:
 
   return input_device.type == "cuda"
 
+def _move_xla_cuda_tensor_to_cuda(tensor):
+  assert tensor.device.type == "xla", "The tensor is not an XLA tensor"
+  is_xla_cuda = True if xu.getenv_as("PJRT_DEVICE", str, "").lower() == "cuda" else False
+  assert is_xla_cuda, "The XLA tensor is not on CUDA"
+  # consumer is torch, producer is torch_xla
+
+  # Similar logic as torch.utils.dlpack.from_dlpack
+  # https://github.com/pytorch/pytorch/blob/b0ef363972203b163cddc95e4c6054b8221c2300/torch/utils/dlpack.py#L114-L115
+  device_id = tensor.device.index
+  stream = torch_xla._XLAC._get_stream_for_cuda_device(device_id)
+  stream = 1 if stream == 0 else stream
+  assert stream is None or type(stream) is int
+  external_stream = torch.cuda.ExternalStream(stream)
+  current_stream = torch.cuda.current_stream()
+  if external_stream != current_stream:
+    event = torch.cuda.Event()
+    event.record(current_stream)
+    external_stream.wait_event(event)
+  dlpack = torch_xla_dlpack.to_dlpack(tensor)
+  cuda_tensor = torch.utils.dlpack.from_dlpack(dlpack)
+  return cuda_tensor
 
 # Given an input list, moves the tensors to the given target_device.
 # The output order will be the same as the input. Non tensors will also still
 # be in the list.
 def _maybe_move_tensors_to_device(tensors: tuple,
                                   target_device: torch.device) -> tuple:
-  if not torch.cuda.is_available():
-    return tensors
```

Comment:
Why are we getting rid of this?

Reply:
For moving XLA tensors to CUDA tensors, if torch.cuda is NOT available, it should fail rather than returning XLA tensors.
Similar arguments applies to the other direction.

If we only support the conversion between CUDA and XLA, I think an assert will be more appropriate. Wdyt?

Reply:
As far as I can see, I don't think this function is called unless we are using CUDA tensors (i.e. `torch.cuda.is_available() == True`). So, I guess this wasn't needed in the first place.

Reply:
Exactly.